pkg/ociwclayer/cim/import.go

```diff
@@ -23,21 +23,20 @@ import (
 )
 
 // ImportCimLayerFromTar reads a layer from an OCI layer tar stream and extracts it into
-// the CIM format at the specified path. The caller must specify the parent layers, if
-// any, ordered from lowest to highest layer.
-// This function expects that the layer paths (both the layer that is being imported & the parent layers) are
-// formatted like `.../snapshots/<id>` and the corresponding layer CIMs are located/will be created at
-// `.../snapshots/cim-layers/<id>.cim`. Each CIM file also has corresponding region & objectID files and those
-// files will also be stored inside the `cim-layers` directory.
+// the CIM format at the specified path.
+// `layerPath` is the directory which can be used to store intermediate files generated during layer extraction (and these file are also used when extracting children layers of this layer)
+// `cimPath` is the path to the CIM in which layer files must be stored. Note that region & object files are created when writing to a CIM, these files will be created next to the `cimPath`.
+// `parentLayerCimPaths` are paths to the parent layer CIMs, ordered from highest to lowest, i.e the CIM at `parentLayerCimPaths[0]` will be the immediate parent of the layer that is being extracted here.
+// `parentLayerPaths` are paths to the parent layer directories. Ordered from highest to lowest.
 //
 // This function returns the total size of the layer's files, in bytes.
-func ImportCimLayerFromTar(ctx context.Context, r io.Reader, layerPath string, parentLayerPaths []string) (int64, error) {
+func ImportCimLayerFromTar(ctx context.Context, r io.Reader, layerPath, cimPath string, parentLayerPaths, parentLayerCimPaths []string) (int64, error) {
```

Comment:
It might be conceptually simpler if we model this instead as:

```go
type Layer struct{
    CimPath string
    DirPath string
}

func ImportCimLayerFromTar(ctx context.Context, r io.Reader, layer *Layer, parents []*Layer) (int64, error) {
    ...
}
```

Reply:
Yeah I wanted to do this too, however, I would prefer to do this in a separate PR if you are okay with it. Here is the reasoning behind that. The layer types that we define here are basically the same types that we have already defined in `internal/layers/wcow_parse.go`. Plus, we will have to parse the snapshotter returned mount in the differ code to convert that mount type into this layer type. That code is also already defined in the layers package.
I think we should define these types (& the corresponding code that converts a snapshotter mount into the appropriate layer type) in the containerd snapshotter (or some other related) package. Then we can use that code in the differ to parse the mount type and pass it here. Similary, in our `internal/layers` package we can use those types as backing types for the `WCOWLayers` interface.  What do you think?

Reply:
I agree it makes sense to expose a `ParseCimFSMount` (or similar) from the cimfs snapshotter package in containerd. Then we can use that function both in the containerd applyer (to pass to `ImportCimLayerFromTar`) and in the shim.

I don't think `ParseCimFSMount` should return a type that directly implements `WCOWLayers`, though. That interface should remain internal to the shim. I think it's fine for the parse function to return a type (or several) types that can be used in an internal `WCOWLayers` implementation, though.

Overall, I'm fine with this happening in a separate PR.

Reply:
Yeah I agree that `ParseCimFSMount` shouldn't return a type that implements `WCOWLayers`. I think we can define other types in `layers` package that are derived from types returned by `ParseCimFSMount` and implement `WCOWLayers` interface. 
I will do that in a separate PR.